cpp/src/replace/replace.cu

```diff
@@ -0,0 +1,209 @@
+{      
+   CUDF_EXPECTS(input.head() != nullptr, "Null input data");
+   if(input.size() == 0 || input.head() == nullptr){
+      return make_numeric_column(input.type(), input.size(), ALL_VALID, stream, mr);
+   }   
+   CUDF_EXPECTS(input.type() == data_type(FLOAT32) || input.type() == data_type(FLOAT64), "Expects float or double input");
+
+    // to device. unique_ptr which gets automatically cleaned up when we leave
+   auto device_in = column_device_view::create(input);
+   
+   // ultimately, the output.
+   auto out = make_numeric_column(input.type(), input.size(), ALL_VALID, stream, mr);
+   // from device. unique_ptr which gets automatically cleaned up when we leave.
+   auto device_out = mutable_column_device_view::create(*out);
+
+   // invoke the actual kernel.  
+  experimental::type_dispatcher(input.type(), 
+                                normalize_nans_and_zeros_kernel_forwarder{},
+                                *device_in,
+                                *device_out,
+                                stream);
+
+   return out;                 
+}                                                 
+
+void normalize_nans_and_zeros(mutable_column_view in_out,
+                              cudaStream_t stream)
+{   
+   CUDF_EXPECTS(in_out.head() != nullptr, "Null input data");
+   if(in_out.size() == 0 || in_out.head() == nullptr){
+      return;
+   }
+   CUDF_EXPECTS(in_out.type() == data_type(FLOAT32) || in_out.type() == data_type(FLOAT64), "Expects float or double input");
+
+   // wrapping the in_out data in a column_view so we can call the same lower level code.
+   // that we use for the non in-place version.
+   column_view input = in_out;
+
+   // to device. unique_ptr which gets automatically cleaned up when we leave
+   auto device_in = column_device_view::create(input);
+
+   // from device. unique_ptr which gets automatically cleaned up when we leave.   
+   auto device_out = mutable_column_device_view::create(in_out);
+
+    // invoke the actual kernel.  
+   cudf::experimental::type_dispatcher(input.type(), 
+                                       normalize_nans_and_zeros_kernel_forwarder{},
+                                       *device_in,
+                                       *device_out,
+                                       stream);
+}
+
+}  // namespace detail
+
+/**
+ * @brief Function that converts inputs from `input` using the following rule
+ *        rule:   Convert  -NaN  -> NaN
+ *                Convert  -0.0  -> 0.0
+ *
+ * @param[in] column_device_view representing input data
+ * @param[in] device_memory_resource allocator for allocating output data 
+ *
+ * @returns new column
+ */
+std::unique_ptr<column> normalize_nans_and_zeros( column_view input,                                                                                                    
+                                                  rmm::mr::device_memory_resource *mr)
+{
```

Comment:
This function should just make a copy of `input` and then invoke the in-place `normalize_nans_and_zeros`.

Reply:
Fixed in ddd408a26f3ada45a156fe5c9df6d8a3746dbd23